canister/src/lib.rs

```diff
@@ -93,18 +94,21 @@ pub fn get_current_fee_percentiles(
 ) -> Vec<MillisatoshiPerByte> {
     verify_api_access();
     verify_network(request.network.into());
+    verify_fully_synced();
     api::get_current_fee_percentiles()
 }
 
 pub fn get_balance(request: GetBalanceRequest) -> Result<Satoshi, GetBalanceError> {
     verify_api_access();
     verify_network(request.network.into());
+    verify_fully_synced();
```

Comment:
I wonder, should we add a flag in the `Config`, something like `disable_api_if_not_fully_synced: Flag`, that enables/disables this check? I think that'd be useful in case we find issues with this check in production and need a quick way to disable it.

Reply:
[Here](https://github.com/dfinity/bitcoin-canister/pull/156) is the follow up with change